apps/web/src/pages/integrations/components/multi-provider/UpdateProviderSidebar.tsx

```diff
@@ -154,6 +161,13 @@ export function UpdateProviderSidebar({
     update({ hideDefaultLauncher: false });
   };
 
+  useEffectOnce(() => {
+    openSelectPrimaryIntegrationModal({
+      environmentId: selectedProvider?.environmentId,
+      channelType: selectedProvider?.channel,
+    });
+  }, hasToSelectPrimaryProvider && !!selectedProvider);
```

Comment:
show the modal when we have a `hasToSelectPrimaryProvider` flag set and the integration is loaded